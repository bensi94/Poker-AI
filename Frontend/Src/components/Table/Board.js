import React from 'react';
import PropTypes from 'prop-types';

import Card from './Card';
import Stack from './Stack';

class Board extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        let cards = [];

        this.props.board.forEach(cardString => {
            cards.push(<Card key={cardString} card={cardString} />);
        });
        return (
            <div className="board-wrapper">
                {cards}
                <Stack tableSize={this.props.tableSize}
                    // If the there is nothing in the pot (pot is undefined), the pot is 0
                    stackAmount={(this.props.pot ? this.props.pot : 0)}
                    left={50}
                    top={90}
                />
            </div>
        );
    }
}

Board.propTypes = {
    board: PropTypes.arrayOf(PropTypes.string),
    tableSize: PropTypes.number.isRequired,
    pot: PropTypes.number
};

export default Board;
