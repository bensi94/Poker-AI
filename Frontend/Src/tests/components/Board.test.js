import React from 'react';
import { shallow } from 'enzyme';

import Board from '../../components/Table/Board';

describe('Board Test sutie', () => {
    let wrapper;

    it('Should render component correctly', () => {
        wrapper = shallow(<Board tableSize={1000} board={['Ah', 'As', '9d', '6c', '7c']} pot={99999}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render correct amount of cards', () => {
        wrapper = shallow(<Board tableSize={1000} board={['Ah', 'As', '9d', '6c']} pot={99999} />);
        expect(wrapper.find('Card').length).toBe(4);
    });

    it('Should render in correct order', () => {
        wrapper = shallow(<Board tableSize={1000} board={['Ah', 'As', '9d', '6c', '7c']} pot={99999} />);
        expect(wrapper.find('Card').at(0).prop('card')).toBe('Ah');
        expect(wrapper.find('Card').at(1).prop('card')).toBe('As');
        expect(wrapper.find('Card').at(2).prop('card')).toBe('9d');
        expect(wrapper.find('Card').at(3).prop('card')).toBe('6c');
        expect(wrapper.find('Card').at(4).prop('card')).toBe('7c');
    });

    it('Should render in correct stackAmount', () => {
        wrapper = shallow(<Board tableSize={1000} board={['Ah', 'As', '9d', '6c', '7c']} pot={99999} />);
        expect(wrapper.find('Stack').first().prop('stackAmount')).toBe(99999);
    });
    it('Should render in correct stackAmount on undefined', () => {
        wrapper = shallow(<Board tableSize={1000} board={['Ah', 'As', '9d', '6c', '7c']} />);
        expect(wrapper.find('Stack').first().prop('stackAmount')).toBe(0);
    });
});
