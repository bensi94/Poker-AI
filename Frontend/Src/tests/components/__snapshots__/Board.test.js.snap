// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Test sutie Should render component correctly 1`] = `
<div
  className="board-wrapper"
>
  <Card
    card="Ah"
    key="Ah"
  />
  <Card
    card="As"
    key="As"
  />
  <Card
    card="9d"
    key="9d"
  />
  <Card
    card="6c"
    key="6c"
  />
  <Card
    card="7c"
    key="7c"
  />
  <Stack
    left={50}
    stackAmount={99999}
    tableSize={1000}
    top={90}
  />
</div>
`;
